log/sockstatlog/logger.go

```diff
@@ -173,6 +189,33 @@ func (l *Logger) poll() {
 	}
 }
 
+// logEvents reads events from the event channel at logInterval and logs them to disk.
+// This method does not return.
+func (l *Logger) logEvents() {
+	enc := json.NewEncoder(l)
+	flush := func() {
```

Comment:
Any reason why is in a separate function? The `poll()` ticker handling is inline in the `select`.

Reply:
It was just for readability.  I originally had it inline, and it had nested for loops that I need to break out of when the event channel was empty.  I thought this was a little easier to reason about what was happening (and less error prone).